### PR TITLE
Make compatible with MW 1.25+

### PR DESCRIPTION
**Pull request by llbraughler**
This makes the extension compatible with MW 1.25+ by using the new extension registration system via JSON, as well as internationalisation using i18n/ json files. Proper handling of plurals was added to the messages.

### CODE REVIEW

File: SpecialUserScore_body.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            return array( 'value' );
+    }
+
+    function formatResult( $skin, $result ) {
+            global $wgContLang;
+
+            $title = Title::makeTitle( NS_USER, $result->title );
+            $userLink = Linker::linkKnown($title, $title->getText() );
+
+            $pages = $this->msg( 'userscore-pages' )->params( $result->page_value )->parse();
```
Comment (nicjansma) [inline SpecialUserScore_body.php]: It doesn't seem like this is used now?

**Comment by nicjansma** [comment]
Thanks for doing this, this is great!

Is the `*_body.php` format a common convention, or can main code just stay in `SpecialUserScore.php`?

**Comment by llbraughler** [comment]
Yep, *_body.php is the standard according to https://www.mediawiki.org/wiki/Manual:Developing_extensions

Also, UserScorePage.php was removed since extension registration is now done with the extension.json file, leaving the only PHP code in the body file.

**Comment by nicjansma** [comment]
Cool!

So I think the issue is:

1. `$pages` isn't used, so it and `"userscore-pages"` can be removed
2. This doesn't work in [miser mode](https://www.mediawiki.org/wiki/Manual:$wgMiserMode) because that only caches `$result->value`, not `$result->page_value`.  So I think there should be a check to see if `$result->page_value` is there.  If so, use `"userscore-result-contrib"`.  If not, it should use a new localized message with just `"{{PLURAL:$1|1 edit|$1 edits}}"`

**Comment by llbraughler** [comment]
Thanks for reviewing that. I updated the commit.

**Pull request merged**